software/firmware/europi.py

```diff
@@ -98,6 +98,7 @@ class AnalogueReader:
     """
 
     def __init__(self, pin, samples=DEFAULT_SAMPLES):
+        self._pin_id = pin

```

Comment:
[optional] I think this would be useful outside of this class, so I'd recommend dropping the implied private underscore.

Reply:
done